app/javascript/api/request.js

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios'
```

Comment:
definitely open to other suggestions of file structure/ file naming / function naming around this new file

Reply:
Maybe we could wrap it up into a apiRequest and expose methods like get, post and delete?

Reply:
@akegan could be good to try out @software-project's suggestion here and see if you prefer the result

Reply:
@software-project is this what you're suggesting? Just want to make sure I get it right

```
const apiCall = async (method, path, data) => {
  if (process.env.NODE_ENV === 'test') {
    var err = Error('API should not be called in TEST')
    console.error(err)
    throw err
  }
  try {
    const request = await axios[method](`/api/v1${path}`, data)
    return request.data
  } catch (e) {
    console.warn(e)
    return false
  }
}

export const get = async (path, data) => {
  apiCall('get', path, data)
}

export const post = async (path, data) => {
  apiCall('post', path, data)
}

export const delete = async (path, data) => {
  apiCall('delete', path, data)
}

export const put = async (path, data) => {
  apiCall('put', path, data)
}
```

Reply:
@software-project just another quick ping on this in case this comment got lost in the shuffle of merging so many PRs yesterday.